lib/jobs/generate-tag.js

```diff
@@ -41,8 +41,13 @@ function generateTagJobFactory(
     function GenerateTagJob(options, context, taskId) {
         GenerateTagJob.super_.call(this, logger, options, context, taskId);
 
-        this.nodeId = context.target || options.nodeId;
-        assert.isMongoId(this.nodeId, 'context.target || options.nodeId');
+        if (options.nodes) {
+            this.nodes = options.nodes;
+            assert.arrayOfString(this.nodes, 'options.nodes');
+        } else {
+            this.nodeId = context.target || options.nodeId;
```

Comment:
Yeah, I only named it nodes because I was worried about potential spelling mistakes since nodeId and nodeIds are so similar. But if we switch it to not include a `nodeId` option as you suggest, I might call the option `nodeIds` only.